### ISSUE TITLE
`A7-1-2`:  Qualifier cannot be declared `constexpr`

### ISSUE THREAD

**Issue by jsinglet** [github/codeql-coding-standards#85]
### Affected rules
 - A7-1-2

### Description

False positive reported for the following test case. In the example, below, `a712` must be `constexpr` for `value` to be declared `constexpr`. However, it is not possible to declare parameters (easily) as `constexpr`. The mitigation for this issue is likely that cases where it is not possible to declare the qualifier as `constexpr`. 

### Example
```cpp
class A7_1_2 {
 public:
  constexpr auto GetValue() const noexcept { return value_; }
  void SetValue(int32_t value) noexcept { value_ = value; }
 private:
  int32_t value_;
};
std::ostream &operator<<(std::ostream &os, const A7_1_2 a712) noexcept {
  const auto value = a712.GetValue(); // Variable value could be marked 'constexpr'.
  os << value;
  return os;
}
```

**Comment by lcartey** [comment]
This is a duplicate of #789, and has been fixed by #794.

**Comment by lcartey** [comment]
Note: this is because we previously weren't considering whether the qualifier to a call could be a compile-time constant.
